src/rotator_library/providers/antigravity_validators.py

```
@@ -0,0 +1,224 @@
+            raise ValueError(f"Message at index {i} missing required 'role' field")
+        
+        if 'content' not in message:
+            raise ValueError(f"Message at index {i} missing required 'content' field")
+        
+        # Validate role
+        valid_roles = {'user', 'assistant', 'system'}
+        if message['role'] not in valid_roles:
+            raise ValueError(
+                f"Message at index {i} has invalid role '{message['role']}'. "
+                f"Must be one of: {', '.join(valid_roles)}"
+            )
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)